workload/report/schema.py

```diff
@@ -0,0 +1,460 @@
+from enum import StrEnum, auto
+import json
+from typing import Optional, Any
+
+from pydantic import BaseModel, model_validator
+import yaml
+
+
+class Parallelism(BaseModel):
+    """Accelerator parallelism details."""
+
+    dp: int = 1
+    """Data parallelism level."""
+    tp: int = 1
+    """Tensor parallelism level."""
+    pp: int = 1
+    """Pipeline parallelism level."""
+    ep: int = 1
+    """Expert parallelism level."""
+
+
+class HostAccelerator(BaseModel):
+    """Host accelerator details."""
+
+    model: str
+    """Accelerator model."""
+    memory: float | int
+    """Amount of memory in one accelerator, in GB."""
+    count: int
+    """Number of accelerators."""
+    parallelism: Optional[Parallelism] = None
+    """Parallelism configuration used."""
+    metadata: Optional[Any] = None
+
+
+class Host(BaseModel):
+    """Host hardware details."""
+
+    accelerator: list[HostAccelerator]
+    metadata: Optional[Any] = None
+
+
+class EngineDetails(BaseModel):
+    """Inverence engine details."""
+
+    name: str
+    version: Optional[str] = None
+    args: dict[str, Any]
+    metadata: Optional[Any] = None
+
+
+class Platform(BaseModel):
+    """Software platform details endompassing all inference engines."""
+
+    engine: list[EngineDetails]
+    """Details on inference engines, list corresponds 1:1 with scenario.host.accelerator."""
+    metadata: Optional[Any] = None
```

Comment:
Would the "presets" used by `GAIE` be stored (in native `yaml` format directly under metadata?